.gitpod.yml

```diff
@@ -1,35 +1,48 @@
 # Learn more about this file at https://www.gitpod.io/docs/references/gitpod-yml
 tasks:
   - name: Service Server
-    # Get Web API urls for use in Next build
+    # Copying env.development and replacing 'localhost' wit Gitpod urls
+    # Passing Gitpod urls for use in Next build
     before: |
-      eval $(gp env -e GITPOD_API_URL=$(gp url 3000))
-      eval $(gp env -e GITPOD_WEB_URL=$(gp url 8000))
-      eval $(gp env -e GITPOD_MICROSERVICE_URL=$(gp url 8443))
+      eval $(gp env -e API_URL=$(gp url 3000))
+      eval $(gp env -e WEB_URL=$(gp url 8000))
+      eval $(gp env -e API_HOST=$(gp url 8443))
+
+      sed -r \
+          -e  "s@(.+=)http://localhost:8000(/[^ ]*)*@\1$WEB_URL\2@g" \
+          -e  "s@(.+=)http://localhost:3000(/[^ ]*)*@\1$API_URL\2@g" \
+          -e  "s@(.+=)http://localhost([^:]*)@\1$API_HOST\2@g" \
+          -e  "s@development\.yml@gitpod\.yml@" \
+          config/env.development > .env
     init: |
       npm install
-      docker-compose --env-file config/env.gitpod build
-    # GitPod workspace urls are not stable, we have to look up urls and build again
+      docker-compose --env-file .env build
+    # GitPod workspace urls are not stable, we'd like to force rebuild nginx again with new urls
     command: |
-      docker-compose --env-file config/env.gitpod up -d
-      npm run start
+      docker-compose --env-file .env up -d --no-dep nginx
```

Comment:
Can you explain why you're running `docker-compose` multiple times here?

What if we use `npm run dev` instead of building the nginx-based frontend?

Reply:
What we really need to do is rework our microservices so that we're building and putting in a container registry from CI vs. always building them from scratch.  cc @raygervais for any ideas on how to approach this in a follow-up.

Reply:
See #2264

Reply:
> What if we use npm run dev instead of building the nginx-based frontend?

That's a very good idea. I didn't think of that.

Reply:
> What we really need to do is rework our microservices so that we're building and putting in a container registry from CI.

Current build time is somewhere around 20 minutes, which is unbearable.

Reply:
20 min!?  Yeah, we need to pull cached builds vs. creating our own.

Reply:
@humphd, is there any way I can start all services except the frontend? Or I'd have to list all services by hand.

Reply:
I'm not aware of a way to do this without saying what you want.  What you did is how I'd do it.